lib/db/leveldb_dbinstance.go

```diff
@@ -171,7 +205,26 @@ func (db *Instance) withHave(folder, device, prefix []byte, truncate bool, fn It
 			l.Debugln("unmarshal error:", err)
 			continue
 		}
-		if cont := fn(f); !cont {
+		if !fn(f) {
+			return
+		}
+	}
+}
+
+func (db *Instance) withHaveSequence(folder []byte, startSeq int64, fn Iterator) {
+	t := db.newReadOnlyTransaction()
+	defer t.close()
+
+	dbi := t.NewIterator(util.BytesPrefix(db.sequenceKey(folder, 0)[:keyPrefixLen+keyFolderLen]), nil)
```

Comment:
We could generate the proper range here directly instead of seeking, it ought to be from `sequenceKey(folder, startSet)` to `sequenceKey(folder, maxInt64)` where `maxInt64 = 1<<63-1`?